server/app/repository/ApplicationRepository.java

```diff
@@ -198,8 +199,7 @@ private Application createOrUpdateDraftApplicationInternal(Applicant applicant,
               .eq("lifecycle_stage", LifecycleStage.DRAFT)
               .findOneOrEmpty();
       Application application =
-          existingDraft.orElse(new Application(applicant, program, LifecycleStage.DRAFT));
-      application.setApplicantData(applicant.getApplicantData());
+          existingDraft.orElseGet(() -> new Application(applicant, program, LifecycleStage.DRAFT));
```

Comment:
Not part of the bugfix, but no need to instantiate an `Application` if it won't be used